lib/server.js

```diff
@@ -1143,19 +1143,37 @@ Server.prototype._setupRequest = function _setupRequest(req, res) {
     // the connection (or we lose the connection).
     // we consider a closed request as flushed from metrics point of view
     function onClose() {
-        res._flushed = true;
-        req._timeFlushed = process.hrtime();
-
         res.removeListener('finish', onFinish);
         res.removeListener('error', onError);
+
         req._connectionState = 'close';
 
+        // Do no handle close if request is already flushed
```

Comment:
Typo "Do not"?

Reply:
Also, is it "if response is already flushed" here, or do we mean "request" in the sense of "the whole request/response"? I think it'd help to distinguish between request and response here for clarity, and in this case it seems we're checking if the _response_ is flushed.